Doc/c-api/unicode.rst

```diff
@@ -1396,18 +1396,28 @@ They all return ``NULL`` or ``-1`` if an exception occurs.
    :c:func:`PyErr_Occurred` to check for errors.
 
 
-.. c:function:: int PyUnicode_EqualToUTF8(PyObject *unicode, const char *string)
+.. c:function:: int PyUnicode_EqualToUTF8AndSize(PyObject *unicode, const char *string, Py_ssize_t size)
 
-   Compare a Unicode object with a UTF-8 or ASCII encoded C string
-   and return true (``1``) if they are equal, or false (``0``) otherwise.
-   If the Unicode object contains null or surrogate characters or
+   Compare a Unicode object with a char buffer which is interpreted as
+   being UTF-8 or ASCII encoded and return true (``1``) if they are equal,
+   or false (``0``) otherwise.
+   If the Unicode object contains surrogate characters or
    the C string is not valid UTF-8, false (``0``) is returned.
```

Comment:
Would you mind to explicit the behavior on null characters and null bytes?

Reply:
What should be added?

Reply:
I'm not sure :-( I propose:

```
While :c:func:`PyUnicode_EqualToUTF8` returns false if the Unicode
object contains null characters, :c:func:`PyUnicode_EqualToUTF8AndSize`
does not treat null characters/bytes differently than other characters/bytes:
*unicode* can contain null characters and *string* can contain null bytes.
```

Reply:
Is it the first C API that we have which treat null characters/bytes differently between the null-terminated flavor and the WithSize() flavor?

Reply:
Some C API which returns C strings, treat them differently -- if it can return size, embedded null characters/bytes are allowed, it it cannot, the returned string is null-terminated, and embedded nulls are forbidden. It was not always so, it is a latter fix for security.

C API which takes C strings, usually do not treat null characters/bytes specially, unless the underlying layer has limitations. And `PyUnicode_EqualToUTF8()` also do not treat them specially. It is just that the first null in the C string is terminating, so null in the Unicode object cannot match.

Reply:
You're right. Maybe just ignore my point and leave the doc as it is. I mark this discusssion as resolved.